### PR TITLE
lnrpc+mobile: use docker to compile/format protos

### PR DESCRIPTION
**Pull request by guggero**
Compiling the protobuf files requires the user to install additional tools and libraries with the correct versions.
Moving to a docker assisted build process makes it easier for the developer to run the compilation.

A docker based build step takes slightly longer the first time but offers a few advantages in the long run:
 - No need for manual version pinning of certain gRPC libraries, the image contains everything.
 - Makes it possible to use different versions of the gRPC libraries in different projects without needing to manually switch.
 - Can easily be extended to generate protobuf stubs for other/all supported programming languages.

Inspired by https://github.com/decred/dcrlnd/pull/101, thanks @matheusd.

### CODE REVIEW

**Comment by halseth** [comment]
This does require the user to have docker installed/running. Is that okay? (my guess is yeah, since most users will not have to compile re-create protos, so for just building lnd from source it won't be needed)

**Comment by bhandras** [comment]
> This does require the user to have docker installed/running. Is that okay? (my guess is yeah, since most users will not have to compile re-create protos, so for just building lnd from source it won't be needed)

Note that `gen_protos.sh` is still there and usable. My opinion is that it's very much acceptable that contributors start using docker in their normal workflows.

File: lnrpc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM golang:1.15.6-buster
+
+RUN apt-get update && apt-get install -y \
+  git \
+  protobuf-compiler='3.6.1*' \
+  clang-format='1:7.0*'
+
+ARG PROTOC_GEN_VERSION
```
Comment (bhandras) [inline lnrpc/Dockerfile:8]: Consider providing defaults for these args.

**Review by bhandras — APPROVED** [review]
tACK, LGTM :100:

**Review by halseth — APPROVED** [review]
LGTM ✅

**Comment by halseth** [comment]
Needs rebase.

**Comment by guggero** [comment]
Rebased.

File: lnrpc/Dockerfile (hunk 1 above)
Comment (guggero) [reply lnrpc/Dockerfile:11]: I explicitly didn't want to add any default values to make sure they're always set. Added a comment to explain.

**Comment by guggero** [comment]
@Roasbeef this PR is ready to be merged but requires admin intervention because some of the required checks were removed/renamed (mobile compilation and RPC compilation check is now combined).

**Pull request merged**